treeherder/intermittents_commenter/commenter.py

```diff
@@ -0,0 +1,285 @@
+import json
+import re
+import time
+from collections import Counter
+from datetime import (date,
+                      datetime,
+                      timedelta)
+
+import requests
+from django.conf import settings
+from django.db.models import Count
+from jinja2 import Template
+from requests.exceptions import RequestException
+
+from treeherder.intermittents_commenter.constants import *
+from treeherder.model.models import (BugJobMap,
+                                     Push)
+from treeherder.webapp.api.utils import get_repository
+
+
+class Commenter(object):
+    """handles fetching, transforming and submitting bug comments based on
+    daily or weekly thresholds and date range; if in dry_run, comments
+    will be output to stdout rather than submitted to bugzilla"""
+
+    def __init__(self, **kwargs):
+        """booleans args are passed from the run_intermittents_commenter
+           based on --weekly and --dry-run flags or from the run_commenter
+           celery task; default values for weekly_mode and dry_run are False."""
+
+        self.weekly_mode = kwargs['weekly_mode']
+        self.dry_run = kwargs['dry_run']
+        self.session = self.new_request()
+
+    def run(self):
+        startday, endday = self.calculate_date_strings(self.weekly_mode, 6)
+        alt_startday, alt_endday = self.calculate_date_strings(True, DISABLE_DAYS)
+        all_params = self.create_comments(startday, endday, alt_startday, alt_endday)
+        self.print_or_submit_comments(all_params)
+
+    def create_comments(self, startday, endday, alt_startday, alt_endday):
+        """Posts a bug comment containing stats to each bug whose total number of
+           occurrences (daily or weekly) meet the appropriate threshold."""
+
+        bug_stats = self.get_bug_stats(startday, endday)
+        alt_bug_stats = self.get_bug_stats(alt_startday, alt_endday)
+        test_run_count = self.get_test_runs(startday, endday)
+
+        all_params = []
+        template = Template(self.open_file('comment.template', False))
+
+        if self.weekly_mode:
+            top_bugs = [bug[0] for bug in sorted(bug_stats.items(), key=lambda x: x[1]['total'],
+                        reverse=True)][:TOP_BUGS_THRESHOLD]
+
+        for bug_id, counts in bug_stats.iteritems():
```

Comment:
Could you switch to the `six` `iter*` methods please (for Python 3 compatibility, for parity with the rest of the casebase). See:
https://github.com/mozilla/treeherder/pull/3418/commits/6e52a6b27cdf5f4950a6110ad0e5e5cf818bb033